experiments/experiments-impl/src/main/java/com/duckduckgo/experiments/impl/VariantManagerImpl.kt

```diff
@@ -106,12 +106,20 @@ class VariantManagerImpl @Inject constructor(
         if (entity.key == "sc" || entity.key == "se") {
             return { isSerpRegionToggleCountry() }
         }
-        if (entity.localeFilter.isEmpty()) {
-            return { noFilter() }
+
+        var matchesLocaleFilter = true
+        var matchesAndroidVersionLocale = true
+
+        if (entity.localeFilter.isNotEmpty()) {
+            val userLocale = Locale.getDefault()
+            matchesLocaleFilter = entity.localeFilter.contains(userLocale.toString())
+        }
+        if (entity.androidVersionFilter.isNotEmpty()) {
+            val userAndroidVersion = appBuildConfig.sdkInt.toString()
+            matchesAndroidVersionLocale = entity.androidVersionFilter.contains(userAndroidVersion)
         }
 
-        val userLocale = Locale.getDefault()
-        return { entity.localeFilter.contains(userLocale.toString()) }
+        return { matchesLocaleFilter && matchesAndroidVersionLocale }
```

Comment:
High-level I don't think this scales nicely. I think is ok for now but we need to think this through. cc/ @aitorvs

Reply:
yes, it doesn't. We'll need to include every filter here when added. When adding next filter I can probably create an array of filters and just check if all of them are true at this point.

Reply:
I changed the implementation to scale better in the end